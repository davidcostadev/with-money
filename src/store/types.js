export const ADD_USER = 'ADD_USER';
export const TO_CURRENT_MONTH = 'TO_CURRENT_MONTH';
export const TO_PREVIUS_MONTH = 'TO_PREVIUS_MONTH';
export const TO_NEXT_MONTH = 'TO_NEXT_MONTH';
