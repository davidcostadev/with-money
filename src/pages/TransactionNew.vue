<template>
  <div>
    <md-toolbar>
      <router-link to="/">
        <md-button class="md-icon-button">
          <md-icon>keyboard_backspace</md-icon>
        </md-button>
      </router-link>
      <h3 class="md-title">Add new Transaction</h3>
    </md-toolbar>
    <transaction-form
      :transactionDate="state_month.format()"
      :onSave="onSave"
    />
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import TransactionForm from '../components/forms/Transaction';
import Transaction from '../services/transactions';
// import Accounts from '../services/accounts';

export default {
  components: {
    TransactionForm,
  },
  // data() {
  //   return {
  //     // name: '',
  //     // value: '',
  //     // AccountId: 0,
  //     // type: 'out',
  //     // isPaid: false,
  //     // transactionDate: new Date(),
  //     // accounts: [],
  //   };
  // },
  computed: {
    ...mapGetters(['state_month']),
  },
  methods: {
    async onSave(data) {
      const transaction = await Transaction.create(data);

      if (transaction.id) {
        this.$store.dispatch('showFlashMessage', 'Transanção salva com sucesso!');
      }

      this.$router.push('/');
    },
  },
  mounted() {
    // this.transactionDate = this.state_month.format();
  },
};
</script>
