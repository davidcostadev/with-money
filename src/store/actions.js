import * as types from './types';

export default {
  addUser({ commit }, payload) {
    commit(types.ADD_USER, payload);
  },
  toNextMonth({ commit }) {
    commit(types.TO_NEXT_MONTH);
  },
  toCurrentMonth({ commit }) {
    commit(types.TO_CURRENT_MONTH);
  },
  toPreviousMonth({ commit }) {
    commit(types.TO_PREVIUS_MONTH);
  },
};
