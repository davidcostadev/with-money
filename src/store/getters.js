export default {
  user(state) {
    return state.user;
  },
  current_month(state) {
    return state.current_month;
  },
  state_month(state) {
    return state.state_month;
  },
  state_month_str(state) {
    return state.state_month_str;
  },
  next_month_str(state) {
    return state.next_month_str;
  },
  previous_month_str(state) {
    return state.previous_month_str;
  },
};
