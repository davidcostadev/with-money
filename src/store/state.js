import Moment from 'moment';

export default {
  user: null,
  current_month: Moment(),
  state_month: Moment(),
  state_month_str: Moment().format('MMM YY'),
  next_month_str: Moment().add(1, 'month').format('MMM YY'),
  previous_month_str: Moment().subtract(1, 'month').format('MMM YY'),
  flash_message: '',
};
