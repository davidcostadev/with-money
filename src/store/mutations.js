import Moment from 'moment';
import * as types from './types';
import { updateStatesMont } from '../utils/state';

export default {
  [types.ADD_USER](state, payload) {
    state.user = payload;
  },
  [types.TO_CURRENT_MONTH](state) {
    state.state_month = state.current_month;
    updateStatesMont(state);
  },
  [types.TO_PREVIUS_MONTH](state) {
    state.state_month = Moment(state.state_month).subtract(1, 'month');
    updateStatesMont(state);
  },
  [types.TO_NEXT_MONTH](state) {
    state.state_month = Moment(state.state_month).add(1, 'month');
    updateStatesMont(state);
  },
};
