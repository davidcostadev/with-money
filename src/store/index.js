import Vue from 'vue';
import Vuex from 'vuex';
import Moment from 'moment';
import mutations from './mutations';
import actions from './actions';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    user: null,
    current_month: Moment(),
    state_month: Moment(),
    state_month_str: Moment().format('MMMM YY'),
    next_month_str: Moment().add(1, 'month').format('MMMM YY'),
    previous_month_str: Moment().subtract(1, 'month').format('MMMM YY'),
  },
  getters: {
    user(state) {
      return state.user;
    },
    current_month(state) {
      return state.current_month;
    },
    state_month(state) {
      return state.state_month;
    },
    state_month_str(state) {
      return state.state_month_str;
    },
    next_month_str(state) {
      return state.next_month_str;
    },
    previous_month_str(state) {
      return state.previous_month_str;
    },
  },
  mutations,
  actions,
});
