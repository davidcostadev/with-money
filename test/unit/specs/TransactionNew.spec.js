import { createLocalVue, shallowMount } from '@vue/test-utils';
import VueMaterial from 'vue-material';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import TransactionNew from '../../../src/pages/TransactionNew';
import store from '../../../src/store';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueMaterial);
localVue.use(VueRouter);

describe('TransactionNew Component', () => {
  describe('Render', () => {
    it('default', () => {
      const wrapper = shallowMount(TransactionNew, { store, localVue });

      expect(wrapper.isVueInstance).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
