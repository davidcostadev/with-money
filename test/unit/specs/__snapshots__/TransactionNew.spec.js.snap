// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionNew Component Render default 1`] = `
<div
  class="full-height"
>
  <mdtoolbar-stub
    class="toolbar"
  >
    <router-link-stub>
      <mdbutton-stub
        class="md-icon-button"
      >
        <mdicon-stub>
          keyboard_backspace
        </mdicon-stub>
      </mdbutton-stub>
    </router-link-stub>
     
    <h3
      class="md-title"
    >
      Add new Transaction
    </h3>
  </mdtoolbar-stub>
   
  <div
    class="content-scroll"
  >
    <transactionform-stub />
  </div>
</div>
`;
